source/state_representation/src/space/cartesian/CartesianState.cpp

```diff
@@ -765,30 +765,27 @@ CartesianState CartesianState::operator+(const CartesianState& state) const {
   return result;
 }
 
-CartesianState& CartesianState::operator-=(const CartesianState& state) {
+CartesianState CartesianState::operator-() const {
   // sanity check
   if (this->is_empty()) {
     throw EmptyStateException(this->get_name() + " state is empty");
   }
-  if (state.is_empty()) {
-    throw EmptyStateException(state.get_name() + " state is empty");
-  }
-  if (!(this->get_reference_frame() == state.get_reference_frame())) {
-    throw IncompatibleReferenceFramesException("The two states do not have the same reference frame");
-  }
+  // create a copy of the state
+  CartesianState result(*this);
   // operation on pose
-  this->set_position(this->get_position() - state.get_position());
-  // specific operation on quaternion using Hamilton product
-  Eigen::Quaterniond orientation =
-      (this->get_orientation().dot(state.get_orientation()) > 0) ? state.get_orientation() : Eigen::Quaterniond(
-          -state.get_orientation().coeffs());
-  this->set_orientation(this->get_orientation() * orientation.conjugate());
+  result.set_position(-result.get_position());
+  result.set_orientation(result.get_orientation().conjugate());
```

Comment:
It seems that this change is not correct (some tests are now failing because of that). Taking the conjugate in the `-` operator and then doing the orientation checks in the `+=` operator is not the same as doing the check in the `-=` operator directly